evalml/pipelines/utils.py

```diff
@@ -1466,21 +1470,29 @@ def stack_X(X, series_id_name, time_index, starting_index=None):
         time_index (str): The name of the time index column.
         starting_index (int): The starting index to use for the stacked DataFrame. If None, the starting index
             will match that of the input data. Defaults to None.
+        series_id_values (set): The unique values of a series ID, used to generate the index. If None, values will
+            be generated from X column values. Defaults to None.
 
     Returns:
         pd.DataFrame: The restacked features.
     """
     original_columns = set()
-    series_ids = set()
+    series_ids = series_id_values or set()
     for col in X.columns:
         if col == time_index:
             continue
         separated_name = col.split("_")
         original_columns.add("_".join(separated_name[:-1]))
-        series_ids.add(separated_name[-1])
+        if series_id_values is None:
+            series_ids.add(separated_name[-1])
 
     restacked_X = []
 
+    if len(series_ids) == 0:
+        raise ValueError(
+            "Unable to stack X as X had no exogenous variables and `series_id_values` is None.",
```

Comment:
I think this would be more accurate if it was `no exogenous variables or `series_id_values` is None`

Reply:
Changed it to `"X has no exogenous variables and `series_id_values` is None."` as it's a bit more succinct